frame/babe/src/lib.rs

```diff
@@ -781,12 +781,26 @@ impl<T: Config> frame_support::traits::EstimateNextSessionRotation<T::BlockNumbe
 		T::EpochDuration::get().saturated_into()
 	}
 
+	fn estimate_current_session_progress(_now: T::BlockNumber) -> Option<Percent> {
+		let elapsed = CurrentSlot::get().saturating_sub(Self::current_epoch_start());
+		Some(Percent::from_rational_approximation(
+			*elapsed,
+			T::EpochDuration::get(),
+		))
+	}
+
 	fn estimate_next_session_rotation(now: T::BlockNumber) -> Option<T::BlockNumber> {
 		Self::next_expected_epoch_change(now)
 	}
 
 	// The validity of this weight depends on the implementation of `estimate_next_session_rotation`
-	fn weight(_now: T::BlockNumber) -> Weight {
+	fn estimate_next_session_rotation_weight(_now: T::BlockNumber) -> Weight {
```

Comment:
Hmm why are not returning the weight with the operation, i.e. `fn estimate_next_session_rotation(now: T::BlockNumber) -> (Option<T::BlockNumber>, Weight)`? The signature is nit pretty, but I also use this quite a lot in my recent work.

Assume an implementation of `estimate_next_session_rotation` will return `None` with almost no weight consumed, but `Some(_)` when it uses a lot of weight. 

Essentially, with the current design we're allowing the same degree of flexibility as simply having `CONST WEIGHT: Weight = T::DbWeight::get().reads(3)`.

Reply:
Updated with your suggestion 👍